src/_nebari/utils.py

```diff
@@ -280,3 +281,14 @@ def is_relative_to(self: Path, other: Path, /) -> bool:
 def set_do_environment():
     os.environ["AWS_ACCESS_KEY_ID"] = os.environ["SPACES_ACCESS_KEY_ID"]
     os.environ["AWS_SECRET_ACCESS_KEY"] = os.environ["SPACES_SECRET_ACCESS_KEY"]
+
+
+def set_azure_resource_group_name(
+    project_name: str = "",
+    namespace: str = "",
+    resource_group_name: str = "",
```

Comment:
It seems a little confusing to have a resource_group_name passed in which is not what is returned by the set_azure_resource_group_name.  My preference would be to call the arg `base_resource_group_name`.